src/main/java/de/telran/projectredirectservice/controller/RedirectController.java

```diff
@@ -16,17 +17,20 @@
 @RestController
 public class RedirectController {
     private final RedirectService redirectService;
+    final StatisticsProducerService statisticsProducerService;
 
-    public RedirectController(RedirectService redirectService) {
+    public RedirectController(RedirectService redirectService, StatisticsProducerService statisticsProducerService) {
         this.redirectService = redirectService;
+        this.statisticsProducerService = statisticsProducerService;
     }
 
-
     @GetMapping("short_server/{url_code}")
     public ResponseEntity<String> redirectByShortURL(@PathVariable String url_code) {
-
         Optional<String> longURL = redirectService.getRedirectUrl( url_code );
+
         if (longURL.isPresent()) {
+            // add statistics
+            statisticsProducerService.sendRedirectSat( new StatDto(url_code, longURL.get()) );
```

Comment:
Most likely should belong to the service layer (RedirectService) since this is a part of the business logic.